app/src/androidTest/java/org/dhis2/usescases/syncFlow/SyncFlowTest.kt

```diff
@@ -18,32 +29,196 @@ class SyncFlowTest : BaseTest() {
     @get:Rule
     val ruleTei = ActivityTestRule(TeiDashboardMobileActivity::class.java, false, false)
     @get:Rule
-    val ruleDataSet = ActivityTestRule(EventSummaryActivity::class.java, false, false)
+    val ruleDataSet = ActivityTestRule(DataSetDetailActivity::class.java, false, false)
+    @get:Rule
+    val ruleSearch = ActivityTestRule(SearchTEActivity::class.java, false, false)
+    @get:Rule
+    val ruleEventWithoutRegistration = ActivityTestRule(ProgramEventDetailActivity::class.java, false, false)
+
+    override fun setUp() {
+        super.setUp()
+        setupMockServer()
+    }
 
     @Test
-    @Ignore
-    fun shouldSuccessfullySyncAChangedEvent() {
-        /*
-        * launch in search
-        * search tei
-        * change event
-        * click on sync
-        * check dialog (person to update)
-        * click on sync
-        * */
-
-        //startTeiActivity()
+    @Ignore("check mockserver and calls")
+    fun shouldSuccessfullySyncAChangedTEI() {
+        val teiName =  "Scott"
+        val teiLastName =  "Kelley"
+        val tbVisit = 0
+
+        mockWebServerRobot.addResponse(GET, API_SYSTEM_INFO_PATH, LoginTest.API_SYSTEM_INFO_RESPONSE_OK)
+        mockWebServerRobot.addResponse(POST, SYNC_TEI_PATH, API_SYNC_TEI_OK)
+
+        setupCredentials()
+        prepareTBProgrammeIntentAndLaunchActivity(ruleSearch)
+
+        searchTeiRobot {
+            closeSearchForm()
+            Thread.sleep(4000)
+            clickOnTEI(teiName, teiLastName)
+        }
 
         teiDashboardRobot {
-            clickOnFab()
-            clickOnReferral()
-            clickOnFirstReferralEvent()
-            clickOnReferralOption()
-            clickOnReferralNextButton()
-            checkEventCreatedToastIsShown()
-            checkEventWasCreated(TeiDashboardTest.LAB_MONITORING)
+            clickOnEventWithPosition(tbVisit)
+        }
+
+        eventRobot {
+            clickOnUpdate()
+            pressBack()
+        }
+
+        syncFlowRobot {
+            Thread.sleep(8000)
+            clickOnSyncTei(teiName, teiLastName)
+            clickOnSyncButton()
+            Thread.sleep(4000)
+            checkSyncWasSuccessfully() //sync failed
         }
 
     }
 
-}
\ No newline at end of file
+    @Test
+    fun shouldShowErrorWhenTEISyncFails() {
+        val teiName = "Lars"
+        val teiLastName = "Overland"
+        val labMonitoring = 1
+
+        mockWebServerRobot.addResponse(POST, SYNC_TEI_PATH, API_SYNC_TEI_ERROR)
+
+        setupCredentials()
+        prepareTBProgrammeIntentAndLaunchActivity(ruleSearch)
+
+        searchTeiRobot {
+            closeSearchForm()
+            Thread.sleep(4000)
+            clickOnTEI(teiName, teiLastName)
```

Comment:
same. Check them out in general . With the new fix in search it should not be a problem anymore